### PR TITLE
Home work 5

### PR DESCRIPTION
**Pull request by MikolaiR**
MovieListFragment
Screenshot_1607968097

MovieDetailsFragment
Screenshot_1607968105

### CODE REVIEW

File: app/src/main/java/com/example/androidacademyhomework/UI/MovieDetailsFragment.kt [hunk 1]
```diff
@@ -47,51 +48,68 @@ class MovieDetailsFragment : Fragment() {
...
-    private fun updateDataMovieForTextView(){
-        textViewDetailFragmentAgeLimit.text = getString(R.string.age_limit, movie.ageLimit)
-        textViewDetailFragmentMovieGenre.text = movie.genre
-        ratingBarDetailsFragment.rating = movie.rating.toFloat()
-        textViewDetailFragmentReviews.text = getString(R.string.reviews, movie.reviews)
-        textViewDetailFragmentNameMove.text = movie.movieName
-        textViewDetailFragmentStory.text = movie.story
+    private fun updateDataMovieForView() {
+        textViewDetailFragmentAgeLimit?.text = getString(R.string.age_limit, args.movie.minimumAge)
+        textViewDetailFragmentMovieGenre?.text = TextUtils.join(", ", args.movie.genres.map { it.name })
```
Comment (emartynov) [inline app/src/main/java/com/example/androidacademyhomework/UI/MovieDetailsFragment.kt]:
```suggestion
args.movie.genres.map{ it.name }.joinToString()
```

File: app/src/main/java/com/example/androidacademyhomework/UI/MovieListFragment.kt [hunk 2]
```diff
@@ -47,25 +48,45 @@ class MovieListFragment : Fragment() {
...
+            var listMovies = listOf<Movie>()
+            whenStarted {
+                listMovies = loadMovies(requireContext())
+            }
+            if (listMovies.isNotEmpty()) {
+                (recyclerMovieList?.adapter as? MovieAdapter)?.apply {
+                    bindMovies(listMovies)
+                }
+            } else {
+                Toast.makeText(requireContext(), "Error load movies", Toast.LENGTH_SHORT).show()
```
Comment (emartynov) [inline app/src/main/java/com/example/androidacademyhomework/UI/MovieListFragment.kt]: To strings.

Comment (emartynov) [inline app/src/main/java/com/example/androidacademyhomework/UI/MovieListFragment.kt]: Can you access UI from background thread?

File: app/src/main/java/com/example/androidacademyhomework/adapters/MovieAdapter.kt [hunk 3]
```diff
@@ -35,43 +38,42 @@ class MovieAdapter(private val clickListener: OnRecyclerItemClicked) : RecyclerV
...
+                }
             }
-            textViewAgeLimit.text = context.getString(R.string.age_limit, movie.ageLimit)
-            textViewMovieGenre.text = movie.genre
-            ratingBarListFragment.rating = movie.rating.toFloat()
-            textViewReviews.text = context.getString(R.string.reviews, movie.reviews)
-            textViewNameMove.text = movie.movieName
-            textViewFilmDuration.text = context.getString(R.string.duration_min,movie.movieDuration)
+            textViewAgeLimit.text = context.getString(R.string.age_limit, movie.minimumAge)
+            textViewMovieGenre.text = TextUtils.join(", ", movie.genres.map { it.name })
```
Comment (emartynov) [inline app/src/main/java/com/example/androidacademyhomework/adapters/MovieAdapter.kt]: Same

File: app/src/main/res/layout/fragment_movie_details.xml [hunk 4]
```diff
@@ -16,9 +16,10 @@
         <ImageView
             android:id="@+id/imageViewDetailFragmentTitleBackground"
             android:layout_width="match_parent"
-            android:layout_height="wrap_content"
+            android:layout_height="300dp"
```
Comment (emartynov) [inline app/src/main/res/layout/fragment_movie_details.xml]: Why hardcode?

**Review by emartynov — COMMENTED** [review]
I see that you have all changes for previous homework here.

Please merge the old one, please.

File: app/src/main/java/com/example/androidacademyhomework/UI/MovieDetailsFragment.kt (hunk 1 above)
Comment (MikolaiR) [reply app/src/main/java/com/example/androidacademyhomework/UI/MovieDetailsFragment.kt]: Fixed

File: app/src/main/java/com/example/androidacademyhomework/UI/MovieListFragment.kt (hunk 2 above)
Comment (MikolaiR) [reply app/src/main/java/com/example/androidacademyhomework/UI/MovieListFragment.kt]: fixed

Comment (MikolaiR) [reply app/src/main/java/com/example/androidacademyhomework/UI/MovieListFragment.kt]: No, but here, everything that is not marked as being in the background thread is done in the main. I think so.

File: app/src/main/java/com/example/androidacademyhomework/adapters/MovieAdapter.kt (hunk 3 above)
Comment (MikolaiR) [reply app/src/main/java/com/example/androidacademyhomework/adapters/MovieAdapter.kt]: fixed

File: app/src/main/res/layout/fragment_movie_details.xml (hunk 4 above)
Comment (MikolaiR) [reply app/src/main/res/layout/fragment_movie_details.xml]: fixed

**Review by emartynov — APPROVED** [review]
LGTM

File: app/src/main/java/com/example/androidacademyhomework/UI/MovieListFragment.kt (hunk 2 above)
Comment (emartynov) [reply app/src/main/java/com/example/androidacademyhomework/UI/MovieListFragment.kt]: My mistake - you're launching on the lifecycle scope

**Pull request merged**